collector.go

```diff
@@ -234,6 +235,57 @@ func getPduValue(pdu *gosnmp.SnmpPDU) float64 {
 	}
 }
 
+// parseDateAndTime extracts a UNIX timestamp from an RFC 2579 DateAndTime.
+func parseDateAndTime(pdu *gosnmp.SnmpPDU) float64 {
+	var (
+		v   []byte
+		tz  *time.Location
+		err error
+	)
+	// DateAndTime should be a slice of bytes.
+	switch pduType := pdu.Value.(type) {
+	case []byte:
+		v = pdu.Value.([]byte)
+	default:
+		log.Debugf("invalid DateAndTime type %v", pduType)
+		return 0
```

Comment:
Should we really be returning 0 here?

Reply:
I was debating how to do error handling here. The function that uses this, `pduToSamples`, doesn't really have any error handling.

Reply:
We can either fail the scrape, or return nothing for this metric and log. Returning bad data is not a good option imho.